pkg/db/migrations/init.go

```diff
@@ -0,0 +1,58 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/contiamo/go-base/v2/pkg/queue/postgres"
+	qpostgres "github.com/contiamo/go-base/v2/pkg/queue/postgres"
+	"github.com/sirupsen/logrus"
+)
+
+// NewIniter creates a db init command that will execut the 000_init.sql
+func NewIniter(assets http.FileSystem) func(context.Context, *sql.DB) error {
+	return func(ctx context.Context, db *sql.DB) error {
+		return backoff.Retry(
+			func() error {
+				return initialize(ctx, db, assets)
+			},
+			backoff.WithContext(backoff.NewExponentialBackOff(), ctx),
+		)
+	}
+}
+
+func initialize(ctx context.Context, db *sql.DB, assets http.FileSystem) (err error) {
+	logrus.Info("starting initialize attempt")
+	logger := logrus.WithField("stmt", "init.sql")
+
+	stmt, err := getSQL("000_init.sql", migrations, assets)
+	if err != nil {
+		return fmt.Errorf("can not read init statement: %w", err)
+	}
+
+	_, err = db.ExecContext(ctx, stmt)
+	if err != nil {
+		return fmt.Errorf("init execution failed: %w", err)
+	}
+
+	// setup queue tables ('tasks' and 'schedules') and setup cascading delete for references
+	logger.Info("queue initialize attempt")
+	err = qpostgres.SetupTables(ctx, db, []postgres.ForeignReference{
```

Comment:
We can't put it here, the configuration will be different for each service.

Reply:
🤦 i thought i removed this, i was juggling three copies of this file, sorry